test/src/unit-cppapi-array.cc

```diff
@@ -411,24 +410,71 @@ TEST_CASE(
   schema.set_domain(domain);
   schema.add_attribute(Attribute::create<std::vector<int32_t>>(ctx, "a"));
   Array::create(array_name_1d, schema);
-  Array array(ctx, array_name_1d, TILEDB_WRITE);
-
-  std::vector<int32_t> a, coord = {10, 20, 30};
-  std::vector<uint64_t> a_offset = {0, 0, 0};
 
-  // Test case where backing buffer for vector is non-null, but size is 0.
-  // For bug https://github.com/TileDB-Inc/TileDB/issues/590
   {
+    Array array(ctx, array_name_1d, TILEDB_WRITE);
+
+    std::vector<int32_t> a, coord = {10, 20, 30};
+    std::vector<uint64_t> a_offset = {0, 0, 0};
+
     a.reserve(10);
     a = {};
     Query q(ctx, array, TILEDB_WRITE);
     q.set_layout(TILEDB_GLOBAL_ORDER);
     q.set_coordinates(coord);
     q.set_data_buffer("a", a);
     q.set_offsets_buffer("a", a_offset);
-    REQUIRE_THROWS(q.submit());
+    REQUIRE_NOTHROW(q.submit());
+
+    array.close();
+  }
+
+  {
+    Array array(ctx, array_name_1d, TILEDB_READ);
+
+    std::vector<int32_t> a(3);
+    std::vector<uint64_t> a_offset(3);
+
+    a.reserve(10);
+    a = {};
+    const std::vector<int> subarray = {10, 30};
+    Query q(ctx, array, TILEDB_READ);
+    q.set_layout(TILEDB_GLOBAL_ORDER);
+    q.set_subarray(subarray);
+    q.set_data_buffer("a", a);
+    q.set_offsets_buffer("a", a_offset);
+    q.submit();
```

Comment:
Is it worth checking to be sure status is COMPLETE 
(maybe less important if a_offset vector is not left zero-initialized, but still wouldn't hurt...)?

Reply:
Also, should #elements returned be checked as well? 
Or, are null attributes not counted? (don't know, just asking)

Reply:
And, should there be some mixture of empty and non-empty attributes tested?
Maybe separate test...
mixed null/non-null crossing tile boundaries as well as not crossing tile boundaries?

Reply:
Added a couple more checks and flushed out a small test matrix. I'll file a ticket to include empty attributes in our smoke tests.